Sources/DVR/SessionDataTask.swift

```diff
@@ -42,30 +42,35 @@ final class SessionDataTask: URLSessionDataTask {
     }
 
     override func resume() {
-        let cassette = session.cassette
-
-        // Find interaction
-        if let interaction = session.cassette?.interactionForRequest(request, headersToCheck: headersToCheck) {
-            self.interaction = interaction
-            // Forward completion
-            if let completion = completion {
-                queue.async {
-                    completion(interaction.responseData, interaction.response, nil)
+        
+        if session.recordMode == .none {
```

Comment:
I don't think this works for the general case. The only recordMode in which you don't need to check for a matching interaction is `.all` because that always records. This will skip looking for a matching interaction for `.once` and `.newEpisodes`.

Reply:
This whole block probably needs to be refactored in terms of the four record modes based on whether there is a cassette and whether there is a match:
- .once = YES cassette, NO match => error; NO cassette => record
- .newEpisodes = YES cassette, NO match => record; NO cassette => record
- .none = YES cassette,  NO match => error, NO cassette => error
- all = always record